src/ProjectNeon/Assets/Features/Effects/Events/QueueEffectBehavior.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class QueueEffectBehavior : MonoBehaviour
+{
+    [SerializeField] private BattleState battleState;
+
+    void OnEnable()
+    {
+        BattleEvent.Subscribe<AddEffectToQueue>((effect) => battleState.QueuedEffects.Add(effect.Effect), this);
```

Comment:
In the lambda, the message isn't the effect. Better to use something more terse, or generic, like `x` or `msg` or `e`...

Reading `effect.Effect` is weird.